src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1369,12 +1369,42 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
                         }
                         else
                         {
+                            SourceLocalSymbol sourceLocal;
+                            type = null;
+
+                            if (node.SyntaxTree == localSymbolLocation.SourceTree && (object)(sourceLocal = localSymbol as SourceLocalSymbol) != null &&
+                                sourceLocal.IdentifierToken.Parent?.Kind() == SyntaxKind.Argument)
+                            {
+                                var argument = (ArgumentSyntax)sourceLocal.IdentifierToken.Parent;
+
+                                if (argument.Identifier == sourceLocal.IdentifierToken &&
+                                    argument.Type.IsVar)
+                                {
+                                    // We are referring to an out variable which might need type inference.
+                                    // If it is in fact needs inference, it is illegal to reference it in the same argument list. 
+                                    // Technically, we could support some restricted scenarios (when it is used as another argument in the same list), 
```

Comment:
@AlekseyTs asked me to post this retraction and correction here.

I stand corrected. I was not precisely correct in my assertion.

The correct assertion is that the implementation resulting from this PR is erroneous, in that it results in a specification and implementation that do not agree with each other. Consequently the PR is not complete until that mismatch is either corrected or noted in one of the three ways approved by @jaredpar.

---
In reply to: [64259322](https://github.com/dotnet/roslyn/pull/11493#discussion_r64259322) [](ancestors = 64259322,64256897)